static/js/ducks/spectra.js

```diff
@@ -42,9 +42,14 @@ messageHandler.add((actionType, payload, dispatch, getState) => {
   if (actionType === REFRESH_SOURCE_SPECTRA) {
     const state = getState().spectra;
 
-    if (Object.keys(state).includes(payload.obj_id)) {
-      dispatch(fetchSourceSpectra(payload.obj_id));
-    }
+    Object.entries(state).forEach(([objID, spectra]) => {
+      if (
+        spectra &&
+        spectra[0]?.obj_internal_key === payload.obj_internal_key
+      ) {
```

Comment:
```suggestion
      if (spectra?.[0]?.obj_internal_key === payload.obj_internal_key) {
```
you can just use optional chaining for that